### ISSUE TITLE
macro call site go to definition

### ISSUE THREAD

**Issue by xbz-24** [rust-lang/rust-analyzer#17032]
**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)

**rustc version**: (eg. output of `rustc -V`)

**editor or extension**: (eg. VSCode, Vim, Emacs, etc. For VSCode users, specify your extension version; for users of other editors, provide the distribution if applicable)

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))

**code snippet to reproduce**: 
```rust
// add your code here

```

![image](https://github.com/rust-lang/rust-analyzer/assets/68678258/cae17b03-be68-439e-8d98-9476047024b2)

![image](https://github.com/rust-lang/rust-analyzer/assets/68678258/db2da8fc-0e3f-4b4a-90ef-af1f7f6ecb84)

```rust
mod cs {
        vulkano_shaders::shader! {
            ty: "compute",
            src: r"
            #version 460

            layout(local_size_x = 8, local_size_y = 8, local_size_z = 1) in;

            layout(set = 0, binding = 0, rgba8) uniform writeonly image2D img;

            void main() {
                vec2 norm_coordinates = (gl_GlobalInvocationID.xy + vec2(0.5)) / vec2(imageSize(img));
                vec2 c = (norm_coordinates - vec2(0.5)) * 2.0 - vec2(1.0, 0.0);

                vec2 z = vec2(0.0, 0.0);
                float i;
                for (i = 0.0; i < 1.0; i += 0.005) {
                    z = vec2(
                        z.x * z.x - z.y * z.y + c.x,
                        z.y * z.x + z.x * z.y + c.y
                    );

                    if (length(z) > 4.0) {
                        break;
                    }
                }

                vec4 to_write = vec4(vec3(i), 1.0);
                imageStore(img, ivec2(gl_GlobalInvocationID.xy), to_write);
            }
            ",
        }
    }

    let shader = cs::load(device.clone()).expect("failed to create shader module")
```

would be useful that go to definition takes you to the call site of the macro, right now this tells very little of the inner workings or what is going really right now.

![image](https://github.com/rust-lang/rust-analyzer/assets/68678258/ff7fd41e-52a5-4e89-b53a-dc54d25690d2)

![image](https://github.com/rust-lang/rust-analyzer/assets/68678258/a78c0a94-1023-40bd-84a6-6137db140b14)

Please let me know if someone cares about implementing this.

**Comment by Veykril** [comment]
What rust-analyzer version are you on? (There is a reason for the issue template existing, so it'd be nice if you filled that out...)

**Comment by Xbz-24** [comment]
![image](https://github.com/rust-lang/rust-analyzer/assets/68678258/9a0afcd5-2e0b-47fc-94bf-c94a611574a4)

**Comment by Xbz-24** [comment]
rustc -V
rustc 1.77.1 (7cf61ebde 2024-03-27)

**Comment by Xbz-24** [comment]
![image](https://github.com/rust-lang/rust-analyzer/assets/68678258/90629536-83ab-4826-93b3-0d1bcd5e52b6)

**Comment by Veykril** [comment]
Looks like we fail to resolve the extern prelude in local modules in block modules, it works fine if the `mod cs` is at the top-level (not nested inside a function or anything)

**Comment by Xbz-24** [comment]
can someone fix this shit? it also happens with local modules when its outside of main

**Comment by lnicola** [comment]
Probably fixed in #17251.
